README.md

```diff
@@ -1 +1,155 @@
+- **Service Layer**: Contains business logic, processing data, and interacting with repositories.
+- **Repository Layer**: Handles direct database interactions for querying and managing data.
+- **Security Layer**: Manages authentication and authorisation processes.
+- **DTO Layer**: Transfers data between application layers, including only necessary information.
+- **Mapper Layer**: Converts between object models to ensure data consistency.
+- **Test Layer**: Validates functionality through unit and integration tests.
+
+
+## Database Entities
+
+
+- **Book**: Represents a book with metadata such as title, author, and publication year.
+- **Category**: Classifies books into genres or subjects.
+- **Shopping Cart**: Temporarily holds selected products for potential purchase.
+- **Cart Item**: Specifies a book and its quantity in the shopping cart.
+- **User**: Registered individual with personal details and assigned permissions.
+- **Role**: Defines user permissions, such as USER or ADMIN.
+- **Order**: Records completed transactions, capturing purchase details.
+- **Order Item**: Details specific books purchased in an order.
+
+
+## Access to Endpoints
+
+
+### User
+- 🟩 **POST** `/auth/registration`: User registration.
+- 🟩 **POST** `/auth/login`: User login.
+
+
+### Book
+- 🟨 **GET** `/books`: Retrieve book catalogue.
+- 🟨 **GET** `/books/{id}`: View book details.
+- 🟨 **GET** `/books/search`: Search books.
+- 🟥 **POST** `/books`: Add a new book.
+- 🟥 **PUT** `/books/{id}`: Update book details.
+- 🟥 **DELETE** `/books/{id}`: Delete book.
+
+
+### Category
+- 🟨 **GET** `/categories`: Retrieve categories.
+- 🟨 **GET** `/categories/{id}`: View category.
+- 🟨 **GET** `/categories/{id}/books`: Get books by category.
+- 🟥 **POST** `/categories`: Create a category.
+- 🟥 **PUT** `/categories/{id}`: Update category.
+- 🟥 **DELETE** `/categories/{id}`: Delete category.
+
+
+### Shopping Cart
+- 🟨 **GET** `/cart`: View cart.
+- 🟨 **POST** `/cart`: Add to cart.
+- 🟨 **PUT** `/cart/cart-items/{id}`: Update cart.
+- 🟨 **DELETE** `/cart/cart-items/{id}`: Remove from cart.
+
+
+### Order
+- 🟨 **GET** `/orders`: View orders history.
+- 🟨 **GET** `/orders/{id}/items`: View order items.
+- 🟨 **GET** `/orders/{orderId}/items/{itemId}`: View specific order item.
+- 🟨 **POST** `/orders`: Place order.
+- 🟥 **PATCH** `/orders/{id}`: Update order status.
+
+
+## SQL Database Diagram
+
+
+![Database Diagram]
+
+
+## How to Run the Book Store API
+
+
+1. **Install Prerequisites**: Ensure Docker, Maven, and JDK Development Kit are installed.
+2. **Clone Repository**: Clone the project repository.
+3. **Configure Environment**: Update the `.env` file with necessary DB and Docker variables:
+  ```plaintext
+  MYSQLDB_USER=root 
+   MYSQLDB_ROOT_PASSWORD=root 
+   MYSQLDB_DATABASE=test_db 
+   MYSQLDB_LOCAL_PORT=3307 
+   MYSQLDB_DOCKER_PORT=3306 
+   SPRING_LOCAL_PORT=8088 
+   SPRING_DOCKER_PORT=8080 
+   DEBUG_PORT=5005
+   ```
```

Comment:
redundant space and wrong file type
```suggestion
  ```env
  MYSQLDB_USER=root 
  MYSQLDB_ROOT_PASSWORD=root 
  MYSQLDB_DATABASE=test_db 
  MYSQLDB_LOCAL_PORT=3307 
  MYSQLDB_DOCKER_PORT=3306 
  SPRING_LOCAL_PORT=8088 
  SPRING_DOCKER_PORT=8080 
  DEBUG_PORT=5005
  ```
```